### PR TITLE
[feature] dark theme for code highlighter in the result list 

### LINKED ISSUE #1354: Code highlighter is not suitable for the dark theme
The code highlighter should support dark & light theme of the UI:

grafik

---- 

remark from @dalf on Matrix: the result width / font size ratio is also an issue (IMO).

- First screenshot: Oscar.
- Second screenshot: Simple

grafik

grafik

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

dark theme for code highlighter in the result list / BTW: the layout has been improved slightly

- https://pygments.org/styles/

For the light mode we use the "default" style and this PR add the "lightbulb" style for the dark mode.

## How to test this PR locally?

`make run` and `!scc  hello world python`

image

## Related issues

Closes: https://github.com/searxng/searxng/issues/1354

### CODE REVIEW

File: searx/static/themes/simple/src/less/code.less [hunk 1]
```diff
@@ -1,9 +1,51 @@
...
+
+      &::-moz-selection {
+        background: transparent; /* Gecko Browsers */
+      }
+      margin-right: 8px;
+      text-align: right;
+    }
+
+    span.linenos {
+      color: #64708d;
```
Comment (return42) [inline searx/static/themes/simple/src/less/code.less:32]: Not sure if it is a good idea to custom the line.nummbers color here / but in the "default" style and in "lightbulb" style the color was to dark .. a value of `#64708d;` fits to both styles .. but when we decide to use other pygments style we have to modify or remove the value here.

File: searx/templates/simple/result_templates/code.html [hunk 2]
```diff
@@ -2,12 +2,31 @@
 
 {{ result_header(result, favicons, image_proxify) -}}
 {{- result_sub_header(result) -}}
-{%- if result.content %}{{ result.content|safe }}{% endif %}</p>
+
+{%- if result.content -%}
+  <p class="content">
+    {{- result.content|safe -}}
+</p>
```
Comment (Bnyro) [inline searx/templates/simple/result_templates/code.html]:
```suggestion
  </p>
```

**Comment by Bnyro** [comment]
That's cool, I didn't even know that we have support for searching code snippets :wink:

File: searx/static/themes/simple/src/less/code.less (hunk 1 above)
Comment (Bnyro) [reply searx/static/themes/simple/src/less/code.less:32]: I agree with you on that.
The line numbers are a bit too highlighted for my taste, it'd be cooler to use a gray color for it.

Nevertheless I don't think that's worth doing any extra patches only to change the line number color, for what its worth most people probably won't notice it anyways :)

File: searx/templates/simple/result_templates/code.html (hunk 2 above)
Comment (return42) [reply searx/templates/simple/result_templates/code.html]: fixed / thanks :+1:

**Pull request merged**